lib/syskit/log/zstd_io.rb

```diff
@@ -12,16 +12,31 @@ class ZstdIO
 
             def initialize(io, read: true, write: false)
                 @io = io
-                @tell = 0
-                @buffer = +""
-                @zstd_in = Zstd::StreamingDecompress.new if read
-                @zstd_out = Zstd::StreamingCompress.new if write
+                @mode_read = read
+                @mode_write = write
+                rewind
             end
 
             def path
                 @io.path
             end
 
+            def rewind
+                @io.rewind
+                @tell = 0
+                @buffer = +""
```

Comment:
what's the point of this +?

Reply:
String literals are frozen (read-only), the + makes it read-write